build.gradle

```diff
@@ -75,9 +75,9 @@ task generateGemJarRequiresFile {
                 w << "# AUTOGENERATED BY THE GRADLE SCRIPT. DO NOT EDIT.\n\n"
                     w << "require \'jar_dependencies\'\n"
                     configurations.runtimeClasspath.allDependencies.each {
-                        w << "require_jar(\'${it.group}\', \'${it.name}\', \'${it.version}\')\n"
+                        w << "require \'${it.name}-${it.version}.jar\'\n"
```

Comment:
If we switch to `require` leveraging JRuby ability to load from jars, do we still need the 
```ruby
require 'jar_dependencies'
```

and the `jar-dependencies` gem?

Reply:
This is a good point.. using require instead of require_jar leads to all versions of the jars being loaded.
It means that using require_jar the jar available is the first one that is passed to require_jar. 

```
3.1.0 :001 > require_jar('io.netty', 'netty-common', '4.1.110.Final')
 => true 
3.1.0 :002 > require_jar('io.netty', 'netty-common', '4.1.109.Final')
 => false 
3.1.0 :003 > Java::io.netty.util.Version.identify["netty-common"].artifactVersion
 => "4.1.110.Final" 
```

But with require it's last one that wins:

```
3.1.0 :001 > require 'io/netty/netty-common/4.1.110.Final/netty-common-4.1.110.Final.jar'
 => true 
3.1.0 :002 > require 'io/netty/netty-common/4.1.109.Final/netty-common-4.1.109.Final.jar'
 => true 
3.1.0 :003 > Java::io.netty.util.Version.identify["netty-common"].artifactVersion
 => "4.1.109.Final" 
```

I'll try to keep the use of require_jar but have a shorter path.

Reply:
Opened https://github.com/elastic/logstash/pull/16432 instead, that will allow paths to be longer than 100 characters by using minitar instead of the ruby's TarWriter.
With the introduction of minitar this PR will no longer be necessary.

Reply:
I think that after that PR is merged this could be closed.